cpp/tensorrt_llm/kernels/renormMoeRoutingKernels.cu

```diff
@@ -253,36 +113,59 @@ __global__ void renormMoeRoutingKernel(InputT* routerLogits, OutputT* topkValues
     auto block = cg::this_thread_block();
     auto warp = cg::tiled_partition<WARP_SIZE>(block);
 
-    InputT minScore = InputT{-INFINITY};
+    BaseType minScore = BaseType{-INFINITY};
     for (uint32_t tokenId = warpIdx; tokenId < numTokens; tokenId += warpNum)
     {
         auto scoreOffset = tokenId * numExperts;
         auto outputOffset = tokenId * topK;
-        InputT inputScore[MaxNumExperts / WARP_SIZE];
+
+        BaseType inputScore[MaxNumExperts / WARP_SIZE];
         IdxT inputIndex[MaxNumExperts / WARP_SIZE];
 
-        InputT warpTopKScore[MaxNumTopExperts];
+        BaseType warpTopKScore[MaxNumTopExperts];
         IdxT warpTopKExpertIdx[MaxNumTopExperts];
 
         // Load scores and indices for this warp
         for (uint32_t i = 0; i < MaxNumExperts / WARP_SIZE; ++i)
         {
             auto expertIdx = i * WARP_SIZE + laneIdx;
             inputScore[i]
-                = expertIdx < numExperts ? static_cast<InputT>(routerLogits[scoreOffset + expertIdx]) : minScore;
+                = expertIdx < numExperts ? static_cast<BaseType>(routerLogits[scoreOffset + expertIdx]) : minScore;
             inputIndex[i] = expertIdx;
         }
 
+        if constexpr (DoSoftmaxBeforeTopK)
+        {
+            calcSoftmax(warp, inputScore);
+        }
         // Reduce topK scores and indices for this warp
         reduce_topk::reduceTopK(warp, warpTopKScore, warpTopKExpertIdx, inputScore, inputIndex, minScore);
 
-        // Perform softmax on topK scores
-        auto score = calcSoftmax(warp,
-            laneIdx < topK ? static_cast<float>(warpTopKScore[laneIdx]) : static_cast<float>(minScore), laneIdx, topK);
-        if (laneIdx < topK)
+        // Normalize the scores
+        if constexpr (DoSoftmaxBeforeTopK)
         {
-            topkValues[outputOffset + laneIdx] = static_cast<OutputT>(score);
-            topkIndices[outputOffset + laneIdx] = warpTopKExpertIdx[laneIdx];
+            float sum = float{1.f};
+            if (normTopkProb)
```

Comment:
Is this implementing the naive renormalize? If so I don't think we really need to support that here do we?